src/source/raster_dem_tile_worker_source.js

```diff
@@ -1,31 +1,52 @@
 // @flow
 
 import DEMData from '../data/dem_data';
+import {RGBAImage} from '../util/image';
+import window from '../util/window';
 
 import type Actor from '../util/actor';
 import type {
     WorkerDEMTileParameters,
     WorkerDEMTileCallback,
     TileParameters
 } from './worker_source';
+const {ImageBitmap} = window;
 
 class RasterDEMTileWorkerSource {
     actor: Actor;
     loaded: {[string]: DEMData};
+    offcreenCanvas: OffscreenCanvas;
+    offcreenCanvasContext: CanvasRenderingContext2D;
 
     constructor() {
         this.loaded = {};
     }
 
     loadTile(params: WorkerDEMTileParameters, callback: WorkerDEMTileCallback) {
         const {uid, encoding, rawImageData} = params;
-        const dem = new DEMData(uid, rawImageData, encoding);
 
+        const imagePixels = (ImageBitmap && rawImageData instanceof ImageBitmap) ? this.getImageData(rawImageData) : rawImageData;
+        const dem = new DEMData(uid, imagePixels, encoding);
         this.loaded = this.loaded || {};
         this.loaded[uid] = dem;
         callback(null, dem);
     }
 
+    getImageData(imgBitmap: ImageBitmap): RGBAImage {
+        // Lazily initialize OffscreenCanvas
+        if (!this.offcreenCanvas || !this.offcreenCanvasContext) {
+            this.offcreenCanvas = new OffscreenCanvas(512, 512);
```

Comment:
Why does 512&times;512 suffice?

Reply:
256x256 is probably better I had this initially because thats what i though our tile resolution was.